File: chazm-model/src/main/java/runtimemodels/chazm/model/event/AssignmentEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.relation.Assignment;

/**
 * The {@linkplain AssignmentEventFactory} interface defines the API for constructing {@linkplain AssignmentEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface AssignmentEventFactory {

    /**
     * Constructs an {@linkplain AssignmentEvent}.
     *
     * @param category   the {@linkplain EventType}.
     * @param assignment the {@linkplain Assignment}.
     * @return a {@linkplain AssignmentEvent}.
     */
    AssignmentEvent build(EventType category, Assignment assignment);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/AssignmentEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.relation.Assignment

/**
 * The [AssignmentEventFactory] interface defines the API for constructing [AssignmentEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface AssignmentEventFactory {

    /**
     * Constructs an [AssignmentEvent].
     *
     * @param category   the [EventType].
     * @param assignment the [Assignment].
     * @return a [AssignmentEvent].
     */
    fun build(category: EventType, assignment: Assignment): AssignmentEvent

}
